src/discof/repair/fd_repair_tile.c

```diff
@@ -247,40 +278,100 @@ loose_footprint( fd_topo_tile_t const * tile FD_PARAM_UNUSED ) {
 
 FD_FN_PURE static inline ulong
 scratch_footprint( fd_topo_tile_t const * tile ) {
-
   ulong l = FD_LAYOUT_INIT;
-  l = FD_LAYOUT_APPEND( l, alignof(fd_repair_tile_ctx_t), sizeof(fd_repair_tile_ctx_t)                           );
-  l = FD_LAYOUT_APPEND( l, fd_repair_align(),             fd_repair_footprint()                                  );
-  l = FD_LAYOUT_APPEND( l, fd_forest_align(),             fd_forest_footprint        ( tile->repair.slot_max   ) );
-  l = FD_LAYOUT_APPEND( l, fd_fec_sig_align(),            fd_fec_sig_footprint       ( 20                      ) );
-  l = FD_LAYOUT_APPEND( l, fd_reasm_align(),              fd_reasm_footprint         ( 1 << 20                 ) );
-  l = FD_LAYOUT_APPEND( l, fd_catchup_align(),            fd_catchup_footprint()                                 );
-  l = FD_LAYOUT_APPEND( l, fd_scratch_smem_align(),       fd_scratch_smem_footprint  ( FD_REPAIR_SCRATCH_MAX   ) );
-  l = FD_LAYOUT_APPEND( l, fd_scratch_fmem_align(),       fd_scratch_fmem_footprint  ( FD_REPAIR_SCRATCH_DEPTH ) );
+  l = FD_LAYOUT_APPEND( l, alignof(ctx_t),       sizeof(ctx_t)                                                  );
+  l = FD_LAYOUT_APPEND( l, fd_repair_align(),    fd_repair_footprint   ()                                       );
+  l = FD_LAYOUT_APPEND( l, fd_forest_align(),    fd_forest_footprint   ( tile->repair.slot_max )                );
+  l = FD_LAYOUT_APPEND( l, fd_policy_align(),    fd_policy_footprint   ( FD_NEEDED_KEY_MAX, FD_ACTIVE_KEY_MAX ) );
+  l = FD_LAYOUT_APPEND( l, fd_inflights_align(), fd_inflights_footprint()                                       );
+  l = FD_LAYOUT_APPEND( l, fd_fec_sig_align(),   fd_fec_sig_footprint  ( 20 )                                   );
+  l = FD_LAYOUT_APPEND( l, fd_reasm_align(),     fd_reasm_footprint    ( 1 << 20 )                              );
+  l = FD_LAYOUT_APPEND( l, fd_catchup_align(),   fd_catchup_footprint  ()                                       );
+  l = FD_LAYOUT_APPEND( l, fd_repair_pending_sign_pool_align(), fd_repair_pending_sign_pool_footprint( FD_REPAIR_PENDING_SIGN_REQ_MAX ) );
+  l = FD_LAYOUT_APPEND( l, fd_repair_pending_sign_map_align(),  fd_repair_pending_sign_map_footprint ( FD_REPAIR_PENDING_SIGN_REQ_MAX ) );
```

Comment:
this can prob just be called `signs` if it's local to the tile and doesn't need a pool